staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/monitoring.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conversion
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	latencyBuckets = prometheus.ExponentialBuckets(0.001, 2, 15)
+)
+
+// converterMonitoringFactory holds metrics for all CRD converters
+type converterMonitoringFactory struct {
+	durations   map[string]*prometheus.HistogramVec
```

Comment:
I think this needs to be clearer about the content of this map. The way I understand it: the keys of this map is the CRD names. And the reason this is a HistogramVec is because we want a Histogram for successes and failures. But then there is also a HistogramVec in the `contevrterMonitoring` struct.

Reply:
This is just a map from CRD name to the metric. The one in contevrterMonitoring is the same as this, except for an specific CRD. On creation of contevrterMonitoring, we get the metric from this map.

Reply:
Added a comment. This is a map from a converter name to its metric. Currently it will have only one item for "webhook". Because we are creating crMetric for each CRD, this will allow sharing the same metric between all CRD webhook converters.

Reply:
thanks this makes it clear immediately